File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = {}
        
        for x in s:
            if x not in count:
                count[x] = 1
            else:
                count[x] += 1
                
        for x in t:
            if x not in count:
                return False
            else:
                count[x] -= 1
                
        for x in count:
            if count[x] != 0:
                return False
        return True
        
        
        

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */

var isAnagram = function(s, t) {
    if(s.length != t.length)
        return false
    const stObj = {}
    for(let st of s){
        stObj[st] = stObj[st] ? ++stObj[st] : 1;
    }
    
    for(let ts of t){
        if(!stObj[ts])
            return false
        --stObj[ts];
    }
    
    Object.values(stObj).some(x => {
        if(x)
            return false
    })
    return true
};